src/services/entity_repository.js

```diff
@@ -143,9 +144,12 @@ export default class EntityRepository {
       'metadata.bundleId': bundleId
     };
 
+    const currentTimestamp = getTimestamp();
+
     const update = {
       $set: {
-        'metadata.bundleTransactionHash': txHash
+        'metadata.bundleTransactionHash': txHash,
+        'metadata.bundleUploadTimestamp': currentTimestamp
```

Comment:
I would also add the bundleUploadTimestamp to the bundle itself.  Just 13 lines above:D